reactor-core/src/main/java/reactor/core/publisher/OnNextFailureStrategy.java

```diff
@@ -0,0 +1,274 @@
+		}
+
+		@Override
+		public Throwable process(Throwable error, @Nullable Object value, Context context) {
+			Exceptions.throwIfFatal(error);
+			Throwable iee = new IllegalStateException("STOP strategy cannot process errors");
+			iee.addSuppressed(error);
+			return iee;
+		}
+	};
+
+	OnNextFailureStrategy RESUME_DROP = new ResumeDropStrategy(null);
+
+	final class ResumeStrategy implements OnNextFailureStrategy {
+
+		final Predicate<Throwable>  errorPredicate;
+		final Consumer<Throwable>   errorConsumer;
+		final Consumer<Object>      valueConsumer;
+
+		ResumeStrategy(@Nullable Predicate<Throwable> errorPredicate,
+				Consumer<Throwable> errorConsumer,
+				Consumer<Object> valueConsumer) {
+			this.errorPredicate = errorPredicate;
+			this.errorConsumer = errorConsumer;
+			this.valueConsumer = valueConsumer;
+		}
+
+		@Override
+		public boolean test(Throwable error, @Nullable Object value) {
+			return errorPredicate == null || errorPredicate.test(error);
+		}
+
+		@Override
+		@Nullable
+		public Throwable process(Throwable error, @Nullable Object value, Context context) {
+			if (errorPredicate == null) {
+				Exceptions.throwIfFatal(error);
+			}
+			else if (!errorPredicate.test(error)) {
+				Exceptions.throwIfFatal(error);
+				return error;
+			}
+			try {
+				valueConsumer.accept(value);
+				errorConsumer.accept(error);
+				return null;
+			}
+			catch (Throwable e) {
+				e.addSuppressed(error);
+				return e;
+			}
+		}
+	}
+
+	final class ResumeDropStrategy implements OnNextFailureStrategy {
+
+		final Predicate<Throwable> errorPredicate;
+
+		ResumeDropStrategy(@Nullable Predicate<Throwable> errorPredicate) {
+			this.errorPredicate = errorPredicate;
+		}
+
+		@Override
+		public boolean test(Throwable error, @Nullable Object value) {
+			return errorPredicate == null || errorPredicate.test(error);
+		}
+
+		@Override
+		@Nullable
+		public Throwable process(Throwable error, @Nullable Object value, Context context) {
+			if (errorPredicate == null) {
+				Exceptions.throwIfFatal(error);
+			}
+			else if (!errorPredicate.test(error)) {
+				Exceptions.throwIfFatal(error);
+				return error;
+			}
+			try {
+				if (value != null) {
+					Operators.onNextDropped(value, context);
+				}
+				Operators.onErrorDropped(error, context);
+				return null;
+			}
+			catch (Throwable e) {
+				e.addSuppressed(error);
+				return e;
+			}
+		}
+	}
+
+	final class LambdaOnNextErrorStrategy implements OnNextFailureStrategy {
+
+		private final BiFunction<? super Throwable, Object, ? extends Throwable> delegateProcessor;
+		private final BiPredicate<? super Throwable, Object> delegatePredicate;
+
+		public LambdaOnNextErrorStrategy(
+				BiFunction<? super Throwable, Object, ? extends Throwable> delegateProcessor) {
+			this.delegateProcessor = delegateProcessor;
+			if (delegateProcessor instanceof BiPredicate) {
+				//noinspection unchecked
+				this.delegatePredicate = (BiPredicate<? super Throwable, Object>) delegateProcessor;
```

Comment:
Is it not best to pass these separately? Otherwise using this (at least with lambda's) is a pain.  

Or is there idea, that to pass lambda' you'd use the other version?  The main issue with the other handler is the two consumers rather than a biconsumer.

Reply:
The idea is to use the other versions, yeah. This one is just an adapter, that has additional hidden logic for `BiFunction`s that are also `BiPredicate` (a sneaky way of allowing user-provided equivalents of the `OnNextErrorStrategy` interface without exposing the interface).

Reply:
Ok, so need `BiConsumer` in other version and this is for more advanced custom strategies.